### PR TITLE
More artifacts triggers and tweaks

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Add new xeno artifacts triggers:
- Heat trigger. Put artifact in a very hot environment (100* Celsius or more) to activate it. Alternatively use hot object like lit welder or citrate.
- Electricity trigger. Connect artifact to powered MV (yellow) cables. Alternatively use stun button or multitool. Don't forget about gloves.
- Timer trigger. This one used for debug or admins. It will just activate artifact every 10 seconds. Don't spawn on random artifacts.

Also some minor tweaks:
- Add a new verb for admins to make artifacts constantly active.
- Temperature changing artifact now heat/cool not only their tiles, but also neighbor tiles. In some cases it will result changing room temperature 4 times faster.
- Interaction trigger now reacts on attacks and pulling.
- Updated to latest codebase standards (sealed classes and static using).

In next update I will focus on a proper scanning equipment (probably similar to [goon implementation](https://wiki.ss13.co/Artifact_Research#Working_Artlab)).

**Screenshots**

image

**Changelog**

:cl:
- add: Added more xeno artifact triggers. Test them with hot or electrical objects.
- tweak: Temperature changing artifacts are much deadlier than before. Isolate them as soon as possible.

### CODE REVIEW

File: Content.Server/Power/Events/PowerPulseEvent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+/// <summary>
+///     Invoked on a target entity, when it was pulsed with an energy.
+///     For instance, interacted with an active stun baton.
+/// </summary>
+public sealed class PowerPulseEvent : EventArgs
+{
+    public EntityUid? User;
+    public EntityUid? Used;
+}
```
Comment (mirrorcult) [inline Content.Server/Power/Events/PowerPulseEvent.cs:11]: out of scope (will probably do this myself later) but combining this and `BeingMicrowavedEvent` into a common EMP event would be nice

File: Content.Server/Power/Events/PowerPulseEvent.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Content.Server.Power.Events;
+
+/// <summary>
+///     Invoked on a target entity, when it was pulsed with an energy.
+///     For instance, interacted with an active stun baton.
+/// </summary>
+public sealed class PowerPulseEvent : EventArgs
```
Comment (mirrorcult) [inline Content.Server/Power/Events/PowerPulseEvent.cs:7]:
```suggestion
public sealed class PowerPulseEvent : EntityEventArgs
```

File: Content.Server/Stunnable/StunbatonSystem.cs [hunk 3]
```diff
@@ -200,5 +204,14 @@ private void TurnOn(StunbatonComponent comp, EntityUid user)
...
             comp.Activated = true;
         }
+
+        private void SendPowerPulse(EntityUid target, EntityUid? user, EntityUid used)
+        {
+            RaiseLocalEvent(target, new PowerPulseEvent()
+            {
+                Used = used,
+                User = user
+            });
```
Comment (mirrorcult) [inline Content.Server/Stunnable/StunbatonSystem.cs:214]:
```suggestion
            RaiseLocalEvent(target, new PowerPulseEvent()
            {
                Used = used,
                User = user
            }, false);
```

for just a directed event

File: Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Components/ArtifactHeatTriggerComponent.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public sealed class ArtifactHeatTriggerComponent : Component
+{
+    /// <summary>
+    ///     Minimal surrounding gas temperature to trigger artifact.
+    ///     Around 100 degrees celsius by default.
+    ///     Doesn't affect hot items temperature.
+    /// </summary>
+    [DataField("activationTemperature")]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float ActivationTemperature = 573;
```
Comment (mirrorcult) [inline Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Components/ArtifactHeatTriggerComponent.cs:20]: this is 300 degrees celsius by default, not 100 as suggested by the comment?

File: Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs [hunk 5]
```diff
@@ -1,27 +1,45 @@
...
+            return;
+
+        _artifactSystem.TryActivateArtifact(uid, args.User);
+    }
+
     private void OnInteract(EntityUid uid, ArtifactInteractionTriggerComponent component, InteractHandEvent args)
     {
-        if (args.Handled)
+        if (args.Handled || !args.InRangeUnobstructed())
             return;
```
Comment (mirrorcult) [inline Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs:40]: does interactionsystem not check `InRangeUnobstructed` itself?

File: Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml [hunk 6]
```diff
@@ -27,6 +27,21 @@
...
         - type: RandomArtifactVisualizer
+    - type: PowerConsumer
+      voltage: Medium
+      drawRate: 500
+    - type: NodeContainer
+      nodes:
+        medium:
+          !type:CableDeviceNode
+          nodeGroupID: MVPower
+          # sadly, HVPower and Apc cables doesn't work right now
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml:38]: why not? is there a bug here

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Power/Events/PowerPulseEvent.cs (hunk 1 above)
Comment (Macoron) [reply Content.Server/Power/Events/PowerPulseEvent.cs:11]: Yeah, maybe made EMP event with power variable. So microwave or stun baton just cause little EMP pulse, not enough to fry electronics on airlocks or APC.

File: Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Components/ArtifactHeatTriggerComponent.cs (hunk 4 above)
Comment (Macoron) [reply Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Components/ArtifactHeatTriggerComponent.cs:20]: Yep, its typo

File: Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml (hunk 6 above)
Comment (Macoron) [reply Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml:38]: Yeah, check #6724. HVPower and Apc cables are always has 0 power for some reason.

File: Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs (hunk 5 above)
Comment (Macoron) [reply Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs]: Now yes, it was deleted by recent interaction clean-up

File: Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml (hunk 6 above)
Comment (mirrorcult) [reply Resources/Prototypes/Entities/Objects/Specific/Xenoarchaeology/artifacts.yml:38]: weird, alright

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**